Can Compressed Output be a runtime option?

I understand it has to be compiled in. But can we compile it in, and allow put a setting in the config to determine whether it's enabled or not?  I didn't set the compressed out and am getting compressed output...

Not to mention, tests fail when you set compressed output to 0

Thu Sep 14 18:46:30 UTC 2017 info: 104 packets, 0 active records, 39 records output, 0 alloc fails, 1.7256e+07 bytes/sec, 1.0400e+05 packets/sec, 3.9000e+04 records/sec
error: could not parse argument 
Traceback (most recent call last):
  File "/app/joy/test/run_tests.py", line 165, in <module>
    test()
  File "/app/joy/test/obsidianbox/test_general.py", line 146, in main_general
    test_unix_os()
  File "/app/joy/test/obsidianbox/test_general.py", line 115, in test_unix_os
    validate_general.run()
  File "/app/joy/test/obsidianbox/test_general.py", line 92, in run
    for line in f:
  File "/usr/lib/python2.7/gzip.py", line 464, in readline
    c = self.read(readsize)
  File "/usr/lib/python2.7/gzip.py", line 268, in read
    self._read(readsize)
  File "/usr/lib/python2.7/gzip.py", line 303, in _read
    self._read_gzip_header()
  File "/usr/lib/python2.7/gzip.py", line 197, in _read_gzip_header
    raise IOError, 'Not a gzipped file'
IOError: Not a gzipped file
make: *** [test] Error 1
Makefile:77: recipe for target 'test' failed

--- a/Makefile
+++ b/Makefile
@@ -99,6 +99,7 @@ man: $(DOCDIR)/joy.1
 clean:
 	rm -f cscope.out cscope.files
 	rm -f $(DOCDIR)/joy.txt
+	find . -name '*.pyc' -delete
 	@cd src; $(MAKE) clean
 	@for a in * .*; do if [ -f "$$a~" ] ; then rm $$a~; fi; done;
 

--- a/joyq
+++ b/joyq
@@ -56,6 +56,7 @@ from sleuth import DictStreamDistributionProcessor
 from sleuth import DictStreamSplitProcessor
 from sleuth import DictStreamElementSelectProcessor
 from sleuth import SleuthPredicate
+from sleuth import SleuthFileType
 from sleuth import enrich_tls
 
 """
@@ -96,15 +97,21 @@ class FlowIteratorFromFile(DictStreamIteratorFromFile):
         if self.file_name is sys.stdin:
             self.f = self.file_name
         else:
-            if self.file_name.endswith('.gz'):
-                self.f = gzip.open(self.file_name, 'r')
-            elif self.pcap_loader.is_pcap():
+            if self.pcap_loader.is_pcap():
                 # Run Joy to generate some JSON for use in this script.
                 self.pcap_loader.run()
                 # Open the json file that was just made.
-                self.f = gzip.open(self.pcap_loader.temp_json['file'], 'r')
+                ft = SleuthFileType(self.pcap_loader.temp_json['file'])
+                if ft.is_gz():
+                    self.f = gzip.open(self.pcap_loader.temp_json['file'], 'r')
+                else:
+                    self.f = open(self.pcap_loader.temp_json['file'], 'r')
             else:
-                self.f = open(self.file_name, 'r')
+                ft = SleuthFileType(self.file_name)
+                if ft.is_gz():
+                    self.f = gzip.open(self.file_name, 'r')
+                else:
+                    self.f = open(self.file_name, 'r')
 
 
 class FlowStitchIterator(DictStreamIterator):

--- a/sleuth/sleuth/sleuth.py
+++ b/sleuth/sleuth/sleuth.py
@@ -541,3 +541,19 @@ class SleuthPredicate(object):
             return self.pred.match(flow)
         else:
             return True
+
+
+class SleuthFileType(object):
+    def __init__(self, filename):
+        self.filename = filename
+
+    def is_gz(self):
+        magic = "\x1f\x8b\x08"
+
+        with open(self.filename) as f:
+            data = f.read(len(magic))
+
+            if data.startswith(magic):
+                return True
+
+        return False

--- a/test/obsidianbox/test_general.py
+++ b/test/obsidianbox/test_general.py
@@ -42,6 +42,7 @@ import subprocess
 import gzip
 import json
 from .utils import end_process
+from .utils import FileType
 
 test_params = [[''],
                ['bidir=1'],
@@ -88,13 +89,23 @@ class ValidateGeneral(object):
                 raise RuntimeError("Subprocess Joy failure")
 
             # Make sure everything is valid JSON
-            with gzip.open(self.tmp_output, 'r') as f:
-                for line in f:
-                    try:
-                        json.loads(line)
-                    except ValueError as e:
-                        logger.error("Invalid JSON object, see file %s", self.tmp_output)
-                        raise e
+            ft = FileType(self.tmp_output)
+            if ft.is_gz():
+                with gzip.open(self.tmp_output, 'r') as f:
+                    for line in f:
+                        try:
+                            json.loads(line)
+                        except ValueError as e:
+                            logger.error("Invalid JSON object, see file %s", self.tmp_output)
+                            raise e
+            else:
+                with open(self.tmp_output, 'r') as f:
+                    for line in f:
+                        try:
+                            json.loads(line)
+                        except ValueError as e:
+                            logger.error("Invalid JSON object, see file %s", self.tmp_output)
+                            raise e
 
             self._cleanup_tmp_files()
 

--- a/test/obsidianbox/test_ipfix.py
+++ b/test/obsidianbox/test_ipfix.py
@@ -43,6 +43,7 @@ import logging
 import json
 import gzip
 from .utils import end_process
+from .utils import FileType
 
 
 class ValidateExporter(object):
@@ -104,13 +105,23 @@ class ValidateExporter(object):
             logger.error("Subprocess Joy IPFIX failure")
             raise RuntimeError("Subprocess Joy IPFIX failure")
 
-        with gzip.open(self.tmp_outputs['collect'], 'r') as f:
-            for line in f:
-                try:
-                    flow = json.loads(line)
-                    self.ipfix_flows.append(flow)
-                except:
-                    continue
+        ft = FileType(self.tmp_outputs['collect'])
+        if ft.is_gz():
+            with gzip.open(self.tmp_outputs['collect'], 'r') as f:
+                for line in f:
+                    try:
+                        flow = json.loads(line)
+                        self.ipfix_flows.append(flow)
+                    except:
+                        continue
+        else:
+            with open(self.tmp_outputs['collect'], 'r') as f:
+                for line in f:
+                    try:
+                        flow = json.loads(line)
+                        self.ipfix_flows.append(flow)
+                    except:
+                        continue
 
     def _sniff_pcap(self):
         """
@@ -131,13 +142,23 @@ class ValidateExporter(object):
             logger.error("Subprocess Joy sniffer failure")
             raise RuntimeError("Subprocess Joy sniffer failure")
 
-        with gzip.open(self.tmp_outputs['sniff'], 'r') as f:
-            for line in f:
-                try:
-                    flow = json.loads(line)
-                    self.sniff_flows.append(flow)
-                except:
-                    continue
+        ft = FileType(self.tmp_outputs['sniff'])
+        if ft.is_gz():
+            with gzip.open(self.tmp_outputs['sniff'], 'r') as f:
+                for line in f:
+                    try:
+                        flow = json.loads(line)
+                        self.sniff_flows.append(flow)
+                    except:
+                        continue
+        else:
+            with open(self.tmp_outputs['sniff'], 'r') as f:
+                for line in f:
+                    try:
+                        flow = json.loads(line)
+                        self.sniff_flows.append(flow)
+                    except:
+                        continue
 
     def validate_export_against_sniff(self):
         """

--- a/test/obsidianbox/test_tls.py
+++ b/test/obsidianbox/test_tls.py
@@ -45,6 +45,7 @@ import uuid
 import glob
 from .utils import end_process
 from .utils import ensure_path_exists
+from .utils import FileType
 
 
 # Default globals
@@ -143,13 +144,23 @@ class ValidateTLS(object):
             latest_file = max(base_files, key=os.path.getmtime)
             logger.debug('latest ' + str(version) + ' base file selected ' + str(latest_file))
 
-            with gzip.open(latest_file, 'r') as f:
-                for line in f:
-                    try:
-                        flow = json.loads(line)
-                        flows.append(flow)
-                    except:
-                        continue
+            ft = FileType(latest_file)
+            if ft.is_gz():
+                with gzip.open(latest_file, 'r') as f:
+                    for line in f:
+                        try:
+                            flow = json.loads(line)
+                            flows.append(flow)
+                        except:
+                            continue
+            else:
+                with open(latest_file, 'r') as f:
+                    for line in f:
+                        try:
+                            flow = json.loads(line)
+                            flows.append(flow)
+                        except:
+                            continue
 
     def _run_tls(self):
         for version, flows in self.new_flows.iteritems():
@@ -165,13 +176,23 @@ class ValidateTLS(object):
                 logger.error("Subprocess Joy failure")
                 raise RuntimeError("Subprocess Joy failure")
 
-            with gzip.open(self.tmp_outputs[version], 'r') as f:
-                for line in f:
-                    try:
-                        flow = json.loads(line)
-                        flows.append(flow)
-                    except:
-                        continue
+            ft = FileType(self.tmp_outputs[version])
+            if ft.is_gz():
+                with gzip.open(self.tmp_outputs[version], 'r') as f:
+                    for line in f:
+                        try:
+                            flow = json.loads(line)
+                            flows.append(flow)
+                        except:
+                            continue
+            else:
+                with open(self.tmp_outputs[version], 'r') as f:
+                    for line in f:
+                        try:
+                            flow = json.loads(line)
+                            flows.append(flow)
+                        except:
+                            continue
 
     def compare_new_against_base(self):
         # Load the baseline json into memory

--- a/test/obsidianbox/utils.py
+++ b/test/obsidianbox/utils.py
@@ -70,3 +70,19 @@ def ensure_path_exists(path):
     except OSError:
         if not os.path.isdir(path):
             raise
+
+
+class FileType(object):
+    def __init__(self, filename):
+        self.filename = filename
+
+    def is_gz(self):
+        magic = "\x1f\x8b\x08"
+
+        with open(self.filename) as f:
+            data = f.read(len(magic))
+
+            if data.startswith(magic):
+                return True
+
+        return False

